### ISSUE TITLE
Playback issues when new chunks restart continuity counters

### ISSUE THREAD

**Issue by zhm-nico-nico-ni** [google/ExoPlayer#1978]
![image](https://cloud.githubusercontent.com/assets/15119157/19590428/6085c4fe-97a4-11e6-9c01-eeb67b03f92e.png)

I play a m3u8 with ExoPlayView 2.0.4,  and always meet  frame like the pic. 
But, when Use ExoPlayView 1.5.2 , HLS play fine.

Is something wrong with decoder?

My hls: 
http://video.esx.bigo.sg/asia_live/g1/M03/00/00/Z9Pny1fOjM6IFHeVAAAOjBWRhVwAAAAFgEzVSYAAA6k12.m3u8

**Comment by AquilesCanta** [comment]
We believe the issue is in the media. From what we observed, one of the chunks is not a direct continuation of the previous chunk. That is, the continuity_counters do not follow. This means that the extractor detects a discontinuity, a seek() is called and the issue is observed. The HLS spec [Section 3](https://tools.ietf.org/html/draft-pantos-http-live-streaming-20#section-3) requires consecutive chunks to be in strict continuation:

> Each Media Segment MUST carry the continuation of the encoded from the end of the segment with the previous Media Sequence Number, where values in a series such as timestamps and Continuity Counters MUST continue uninterrupted.  The only exceptions are the first Media Segment ever to appear in a Media Playlist, and Media Segments which are explicitly signaled as discontinuities (Section 4.3.2.3). **Unmarked media discontinuities can trigger playback errors**.

To sum up, to avoid the issue you will either have to add a #EXT-X-DISCONTINUITY tag between the non-contiguous chunks or fix the continuity counters.
